From 09360f51de3d54d3fb8d66c3e8d715d7ba30ee81 Mon Sep 17 00:00:00 2001
From: Jeremy Newton <Jeremy.Newton@amd.com>
Date: Fri, 23 Jun 2023 14:07:21 -0400
Subject: [PATCH 1/5] Fix python script install permissions

The keyword "PROGRAMS" should be used in place of "FILES" in order to
make sure executable scripts have the correct permissions.

Change-Id: I6c287dc1291774ad6d97a04d621957dea0a1b697
Signed-off-by: Jeremy Newton <Jeremy.Newton@amd.com>
---
 rocm_smi/CMakeLists.txt | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/rocm_smi/CMakeLists.txt b/rocm_smi/CMakeLists.txt
index 8401dbe..e43d72e 100755
--- a/rocm_smi/CMakeLists.txt
+++ b/rocm_smi/CMakeLists.txt
@@ -128,11 +128,9 @@ install(FILES ${COMMON_SRC_ROOT}/include/rocm_smi/${ROCM_SMI_TARGET}Config.h
                                         DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/rocm_smi)
 install(FILES ${COMMON_SRC_ROOT}/include/rocm_smi/kfd_ioctl.h
                                         DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/rocm_smi)
-install(FILES ${COMMON_SRC_ROOT}/python_smi_tools/rsmiBindings.py
+install(PROGRAMS ${COMMON_SRC_ROOT}/python_smi_tools/rsmiBindings.py
                                         DESTINATION ${CMAKE_INSTALL_LIBEXECDIR}/${ROCM_SMI})
-install(FILES ${COMMON_SRC_ROOT}/python_smi_tools/rocm_smi.py
-       PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE GROUP_READ
-                         GROUP_WRITE GROUP_EXECUTE WORLD_READ WORLD_EXECUTE
+install(PROGRAMS ${COMMON_SRC_ROOT}/python_smi_tools/rocm_smi.py
                                         DESTINATION ${CMAKE_INSTALL_LIBEXECDIR}/${ROCM_SMI})
 install(FILES ${CMAKE_CURRENT_BINARY_DIR}/bin/rocm-smi
                                         DESTINATION ${CMAKE_INSTALL_BINDIR})
-- 
2.41.0

